src/dialog/index.ts

```diff
@@ -259,6 +280,146 @@ function createDialog(options: IDialogOptions, buttonNodes: HTMLElement[]): HTML
   return node;
 }
 
+/**
+ * Create the dialog node.
+ */
+class WidgetDialog extends Panel {
+  /**
+   *
+   */
+  constructor(options: IDialogOptions, resolve: (value: IButtonItem) => void, reject?: (error: any) => void) {
+    super();
+
+    if (!(options.body instanceof Widget)) {
+      throw 'A widget dialog can only be created with a widget as its body.';
+    }
+
+    this.resolve = resolve;
+    this.reject = reject;
+
+    // Create the dialog nodes (except for the buttons).
+    let content = new Panel();
+    let header = new Widget({node: document.createElement('div')});
+    let body = new Panel();
+    let footer = new Widget({node: document.createElement('div')});
```

Comment:
I'd consider making `header` and `footer` just be regular DOM nodes since they don't really need to be `Widget` instances. But I don't feel strongly about it.